apollo-router/src/spec/schema.rs

```diff
@@ -55,7 +108,37 @@ impl Schema {
             parse(&api_schema, configuration)
         }
 
-        fn parse(schema: &str, _configuration: &Configuration) -> Result<Schema, SchemaError> {
+        fn parse(schema: &str, configuration: &Configuration) -> Result<Schema, SchemaError> {
+            let parser_result = parse_with_apollo_parser(schema, configuration);
+            if cfg!(debug_assertions) {
+                let compiler_result = parse_with_apollo_compiler(schema, configuration);
+                #[allow(clippy::panic)] // only with debug_assertions
+                match (&parser_result, &compiler_result) {
+                    (Ok(a), Ok(b)) => similar_asserts::assert_eq!(a, b),
+                    (Ok(_), Err(SchemaError::Validation(diagnostics))) => {
+                        panic!(
```

Comment:
this should probably be an error log instead of a panic, because we might run into those differences in unrelated issues, and the panics should not block the work

Reply:
My impression is that logs are spammy and nobody reads logs unless they’re looking for something. The intent of this PR is that we notice when this comparison regresses. It’s already under `cfg!(debug_assertions)` so doesn’t run in production.

For local development I understand this could get in the way while working on something else. How about having this panic only on CI? I was tempted to use `failfast_error!` here but some tests currently fail with `--features failfast` so they’d need changes before we can enable that on CI.